### DISCUSSION TITLE
Motion in Mística

### DISCUSSION THREAD

**Discussion by yceballost** [Telefonica/mistica-design#492] [Proposals]
Llevamos mucho tiempo queriendo tener documentado todo lo que está relacionado con animaciones en Mística.

- Principios de animación?
- Cómo aplica el motion a Mística?
- Definir las animaciones y sus tipologías (tipo: lineal, curvas, etc) (velocidad: rapido, normal, lento)
- Cuándo usarla unas u otras
- Definir patrones genéricos de animación que pueden ser aplicados a los diferentes componentes
- Mapear y listar qué componentes son susceptibles de tener animaciones particulares y qué animacioones aplicarían al conjunto de la pantalla.
- Cómo documentarlo

En mi experiencia desarrollando Mística me he dado cuenta que las animaciones muchas veces necesitan de un toque personalizado para según que casos, tiene sentido tener definido X tipos de curvas u tiempos? En otros sistemas de diseño es lo habitual pero yo no sé si soy taaan creyente de eso, aunque entiendo el objetivo.

---

Recursos que he ido encontrando que pueden ser útiles a la hora de definir esta parte, no tenemos por que aplicar cosas como las capturas que he añadido, pero que sirva para tener información.

https://www.designsystems.com/5-steps-for-including-motion-design-in-your-system/
https://medium.com/@aviadtend/motion-design-system-practical-guide-8c15599262fe
https://cultureamp.design/guidelines/animation/

![image](https://user-images.githubusercontent.com/6722153/175564409-7ca23564-9a76-4c5b-bda8-f438bc755b5c.png)
![image](https://user-images.githubusercontent.com/6722153/175564454-363e49e9-903e-4fba-8a33-30a7716702fd.png)
![image](https://user-images.githubusercontent.com/6722153/175564481-9bef9a45-6d5a-4b0e-a7c5-8b81743dc515.png)
![image](https://user-images.githubusercontent.com/6722153/175564503-97cecfa9-1b79-457e-9694-17132fc58a0c.png)

**Comment by yceballost** [comment]
En explore se ha hecho el ejercicio de añadir una animación al scroll que hace aparecer los elementos

[▶️ Video](https://photos.google.com/share/AF1QipMsgKCE45hGzihDrn0ii9AG2qdToBj_01K1H0Hp1lji9sO4ciBkXeAYD0iJUxx3pg/photo/AF1QipNhtK1rwveqTTxyFw3pGhcNHc7Fn5unrHfmLQBC?key=N2R0YVZpNHNPNk0zYi1SSmY3Q211QUV5dUp1RFBn)

- Cuál es el objetivo de esta animación? 
- Se puede hacer muy pesada si se aplica a todo el producto?

**Comment by yceballost** [comment]
Añado este ticket por que en algún momento se habló de definir animaciones para "dismiss" or "slide"
https://jira.tid.es/browse/WEB-290

**Comment by yceballost** [comment]
Creo que estaría bien exponer como componente de animación la animación que existe en el componente Feedback Screen que hace aparecer el title y la description
https://ibit.ly/H_hb

Al igual que existe el `<FadeIn>` podríamos también tener ésta. El nombre? ni idea.
Otra posibilidad es que el propio FadeIn traiga la opción de añadirle el translateY.. pero igual es mezclar churras con merinas

wdyt?

**Comment by paulasuarezb** [comment]
Varios recursos para ayudar a desvelar cómo va a ser el motion en Mística:

- En Material tienen mucha doc sobre motion, incluso sobre multibrands. https://m2.material.io/design/motion/understanding-motion.html#principles
- Fluent 2 -> https://fluent2.microsoft.design/motion
- Decathlon -> https://wolffolins.com/work/decathlon
- Instagram -> https://studiodumbar.com/work/instagram

**Comment by paulasuarezb** [comment]
Más info sobre documentación de motion
- Kinetics https://www.lightningdesignsystem.com/kinetics/overview/
- Accesibilidad en Motion https://www.lightningdesignsystem.com/kinetics/accessible-kinetics/
- https://medium.com/salesforce-ux/what-you-need-to-know-about-the-new-salesforce-kinetics-system-ce10e9ae66a9

**Comment by AnaMontes11** [comment]
Retomo esta discussion porque hemos avanzado en muchos aspectos del sistema, y creo que ahora puede ser un buen momento empujar este tema que llevamos tanto tiempo con ganas de abordar. 

Además, ejemplos recientes como Material 3 Expressive están apostando claramente por motion como una parte esencial para enriquecer la interfaz y reforzar la identidad del producto.

Para romper el hielo, propongo empezar de forma práctica con un mapeo de componentes susceptibles de motion: detectar qué componentes ya tienen alguna animación o cuáles podrían beneficiarse (por ejemplo: botones con loading, drawers, snackbars, modals, switches, tooltips...).

Esto nos permitiría tener una visión clara de:

- dónde ya hay interacción con movimiento,
- dónde podría aportar valor

y desde ahí pensar los principios, definir las animaciones, tipologías etc

Dejo dos referencias: 

- M3 expressive: https://m3.material.io/styles/motion/overview/how-it-works
- MOTION en Carbon: https://carbondesignsystem.com/elements/motion/overview/
